zerver/worker/queue_processors.py

```diff
@@ -219,7 +219,10 @@ def consume(self, data):
 class ConfirmationEmailWorker(QueueProcessingWorker):
     def consume(self, data):
         # type: (Mapping[str, Any]) -> None
-        invitee = get_prereg_user_by_email(data["email"])
+        invitee = PreregistrationUser.objects.filter(id=data["prereg_id"]).first()
+        # The invitation could have been revoked
```

Comment:
Aha! This presumably fixes (and the comment explains) the exceptions we've been seeing where the PreregistrationUser doesn't exist.

It'd be good to make this bugfix clear in the commit history. Ideally it'd be a separate commit. It'd also be OK if this commit message just mentioned that the behavior was changing here to fix this bug.